pkg/blockbuilder/tsdb.go

```diff
@@ -100,6 +101,44 @@ func (b *TSDBBuilder) Process(ctx context.Context, rec *kgo.Record, lastBlockMax
 		}
 	}()
 
+	mapAppendError := func(err error) error {
```

Comment:
also - shouldn't you use this mapping when committing the appender too? I think it might return some similar errors as during  Append

Reply:
I guess any error in Commit() is a hard error. Ingester also treats as a hard error. If there is any out of order sample or any soft error, Commit() ignores it (from top of my head).

Reply:
TIL that the [headAppender.Commit](https://github.com/grafana/mimir/blob/fefa35e43bcc36c58f8baa3b1de0171b5f590b28/vendor/github.com/prometheus/prometheus/tsdb/head_append.go#L797) just never returns any errors